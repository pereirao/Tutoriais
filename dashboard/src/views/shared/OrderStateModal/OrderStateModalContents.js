// @flow

import React, {PureComponent} from 'react';
import I18n from '../../../localization';
import type {Order} from '../../../business/order';
import {order_helpers} from '../../../business/order';
import style_sheet from './OrderStateModal.module.css';
import OrderTypeIcon from '../OrderTypeIcon';
import OrderStateForm from './OrderStateForm';

const {
  deliveryMethodType,
  nextState,
  isGift,
  recipientName,
  orderStateModalIconType
} = order_helpers;

// TODO use MBForm
type OrderStateModalContentsProps = {
  order: Order
};

class OrderStateModalContents extends PureComponent {
  props: OrderStateModalContentsProps

  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {order} = this.props;
    const scope = `ui.modal.${deliveryMethodType(order)}.${nextState(order)}`;
    const order_title = I18n.t(`ui.modal.${isGift(order) ? 'gift' : 'order'}`);

    return (
      <div className={style_sheet.container}>
        <OrderTypeIcon type={orderStateModalIconType(order)}/>
        <h3 className={style_sheet.question}>{I18n.t(`${scope}.question`, {
          recipient_name: recipientName(order),
          order_title
        })}</h3>
        <OrderStateForm
          order_id={order.id}
          show_dsp_flipper={order.show_dsp_flipper}
          delivery_method_type={deliveryMethodType(order)}
          proposed_state={nextState(order)}
          is_gift={isGift(order)}/>
      </div>
    );
  }
}

export default OrderStateModalContents;
