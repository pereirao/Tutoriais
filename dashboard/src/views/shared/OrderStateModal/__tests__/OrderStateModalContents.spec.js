import React from 'react';

import order_factory from '../../../../business/order/__tests__/order.factory';
import OrderStateModalContents from '../OrderStateModalContents';

it('renders correctly', () => {
  expect(shallow(<OrderStateModalContents order={order_factory.build()} />)).toMatchSnapshot();
});
