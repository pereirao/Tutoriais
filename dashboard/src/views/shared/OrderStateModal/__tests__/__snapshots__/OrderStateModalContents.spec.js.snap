// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <OrderTypeIcon
    type="on_demand"
  />
  <h3
    className="question"
  >
    Confirm delivery order for Biggie Smalls - East Coast?
  </h3>
  <Connect(OrderStateForm)
    delivery_method_type="on_demand"
    is_gift={false}
    order_id={1}
    proposed_state="confirmed"
  />
</div>
`;
